app/controllers/authenticate_controller.rb

```diff
@@ -223,6 +231,35 @@ def handle_authentication_error(err)
     end
   end
 
+  def audit_authn_error(err)
+    authenticate_audit_event(err)
+  end
```

Comment:
```suggestion
  def audit_authn_error(err)
    Audit.logger.log(
      authenticate_audit_event(err)
    )
  end
```

Reply:
also - let's make the method name more generic so we have `audit_success` and `audit_failure`
```suggestion
  def audit_failure(err)
```